business_rule_engine/__init__.py

```diff
@@ -61,16 +80,11 @@ def check_condition(self, params, *, set_default_arg=False, default_arg=None):
         self.status = bool(rvalue_condition)
         return rvalue_condition
 
-    def run_action(self, params, *, set_default_arg=False, default_arg=None):
-        action_compiled = self._compile_condition(self.actions)
-        params_actions = self._get_params(params, action_compiled, set_default_arg, default_arg)
-        return action_compiled(**params_actions)
-
-    def execute(self, params, *, set_default_arg=False, default_arg=None) -> Tuple[bool, Any]:
+    async def execute(self, params, *, set_default_arg=False, default_arg=None) -> Tuple[bool, Any]:
```

Comment:
* This will break existing code, because async can not called without await.
* removing the method ``run_action`` can also break existing code, when ``run_action`` is called without the ``execute`` method

**Note:** 

you can use ``check_condition`` and ``run_action`` to implement your own execution rules.
For example, you want to restrict the execution of an action on a condition, which is not part of the rule.

This is useful, if you only want to allow certain users of a web application to run a specific action, but the usercheck should not be part of the rule.